### PR TITLE
[ROCm] Enable flex decoding unit tests

### PR DESCRIPTION
**Pull request by jerrymannil**
Flex decoding tests are passing with upstream pytorch on MI300X/MI2XX.
Only flex attention unit tests have issues.

[result_mi250.log](https://github.com/user-attachments/files/16286954/result_mi250.log)

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by jeffdaily — APPROVED** [review]

**Review by pruthvistony — APPROVED** [review]

**Comment by pruthvistony** [comment]
@jerrymannil ,
Moved the PR to draft state to wait for the CI completion and once CI is green, move out of draft and get upstream approval to merge it.

**Comment by jerrymannil** [comment]
> ## Merge failed
> **Reason**: Approvers from one of the following sets are needed:
> 
> * superuser (pytorch/metamates)
> * Core Reviewers (mruberry, lezcano, Skylion007, ngimel, peterbell10)
> * Core Maintainers (soumith, gchanan, ezyang, dzhulgakov, malfet)
> 
> Details for Dev Infra team

@ezyang  Can you approve ?

**Comment by pruthvistony** [comment]
@malfet ,
Can you please review the PR.

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**
